File: smart-fridge-java-backend/src/main/java/css/ch/smartfridgejavabackend/user/repository/UserRepository.java
package css.ch.smartfridgejavabackend.user.repository;

import css.ch.smartfridgejavabackend.user.model.User;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.Optional;

public interface UserRepository extends MongoRepository<User, String> {
    boolean existsByEmail(String email);

    Optional<User> findUserByEmail(String email);
}

File: smart-fridge-backend/src/main/kotlin/ch/bbzw/smartfridgebackend/user/repository/UserRepository.kt
package ch.bbzw.smartfridgebackend.user.repository

import ch.bbzw.smartfridgebackend.user.model.User
import org.springframework.data.mongodb.repository.MongoRepository
import java.util.Optional

interface UserRepository : MongoRepository<User, String> {
    fun findUserByEmail(email: String): Optional<User>

    fun searchById(id: String?): Optional<User>

    fun existsByEmail(email: String): Boolean
}
